library/forms.inc

```diff
@@ -141,65 +119,57 @@ function getEncounters($pid, $dateStart = '', $dateEnd = '', $encounterRuleType
         $where .= "form_encounter.pid = ?";
         array_push($arraySqlBind, $pid);
     }
-
         $res = sqlStatement("SELECT distinct encounter FROM $from WHERE $where ORDER by date desc;", $arraySqlBind);
-
     for ($iter=0; $row=sqlFetchArray($res); $iter++) {
         $all[$iter] = $row;
     }
-
         return $all;
 }
-
 function getEncounterDateByEncounter($encounter)
 {
     global $attendant_type;
     $table = $attendant_type == 'pid' ? 'form_encounter' : 'form_groups_encounter';
     // $sql = "select date from forms where encounter='$encounter' order by date";
-    $sql = "select date from $table where encounter='$encounter' order by date";
-    return sqlQuery($sql);
+    $sql = "SELECT date from ? where encounter = ? order by date";
```

Comment:
The escaping of encounter is correct, but we need to use a different function for escaping tables:
https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L132
This function is always available and uses whitelisting to essentially escape table names:
```
$sql = "SELECT date from " . escape_table_name($table) . " where encounter = ? order by date";
return sqlQuery($sql, [$encounter]);
```